client/service/src/builder.rs

```diff
@@ -771,13 +772,15 @@ where
 		return Err("Warp sync enabled, but no warp sync provider configured.".into())
 	}
 
-	if config.state_pruning.is_archive() {
-		match config.network.sync_mode {
-			SyncMode::Fast { .. } => return Err("Fast sync doesn't work for archive nodes".into()),
-			SyncMode::Warp => return Err("Warp sync doesn't work for archive nodes".into()),
-			SyncMode::Full => {},
-		};
-	}
+	// FIXME
+	eprintln!("AN ACTUALLY USED STATE_PRUNING IS NEEDED HERE");
```

Comment:
@andresilva @bkchr 

Can there be an appropriate way to access the meta-data of the StateDb from this context?